src/async_wrap.cc

```diff
@@ -122,9 +121,27 @@ void Emit(Environment* env, double async_id, AsyncHooks::Fields type,
 }
 
 
-void AsyncWrap::EmitPromiseResolve(Environment* env, double async_id) {
-  Emit(env, async_id, AsyncHooks::kPromiseResolve,
-       env->async_hooks_promise_resolve_function());
+void AsyncWrap::EmitPromiseEvent(Environment* env,
+                                 double async_id,
+                                 PromiseHookType type,
+                                 Local<Value> value) {
+  AsyncHooks* async_hooks = env->async_hooks();
+
+  if (async_hooks->fields()[type] == 0 || !env->can_call_into_js())
+    return;
+
+  Local<Function> fn = env->async_hooks_promise_event_function();
+
+  v8::HandleScope handle_scope(env->isolate());
```

Comment:
This would technically need to be swapped with the line above (it’s only working because we’re doing an evil hack and look into V8’s object representations to load persistent references faster)